docs/source/programming/SSH-SWC-cluster.md

```diff
@@ -0,0 +1,235 @@
+```
+
+Save the file by pressing `Ctrl+O`, then `Enter`.
+Exit the `nano` editor by pressing `Ctrl+X`.
+
+From now on, you can directly SSH into the *gateway* node by typing:
+```{code-block} console
+$ ssh swc-gateway
+```
+You can also use the same syntax to SSH into the *bastion* node:
+```{code-block} console
+$ ssh swc-bastion
+```
+In both cases, one `logout` command will return you to your local machine.
+
+## SSH keys
+If you are bored of typing your password every time you SSH into the cluster,
+you can set up authentication via SSH keys. You will have to do some work
+upfront, but it will save you tons of time in the long run. Plus, it's more secure.
+
+::: {dropdown} How does SSH key authentication work?
+:color: info
+:icon: info
+You generate a pair of keys locally - a public and a private one -
+and then copy the public key to the remote machine.
+When you try to SSH into the remote machine, the SSH client on your local machine
+will use the private key to generate a signature, which the SSH server on the
+remote machine will verify using the public key. If the signature is valid,
+you will be granted access.
+
+There are several cryptographic algorithms that can be used to generate the keys.
+They can be selected using the `-t` argument of the `ssh-keygen` command.
+In the following example, we use `ed25519`, as it strikes a good balance between
+security and speed for most use cases.
+:::
+
+To generate a pair of SSH keys, run the following command on your local machine:
+```{code-block} console
+$ ssh-keygen -t ed25519
+```
+
+You will be prompted to enter a file path for the key. You may accept the
+default - `~/.ssh/id_ed25519` - or choose another path/name.
+
+Next, you will be prompted to enter a passphrase.
+This is an extra layer of security, but you can leave it blank if you want.
+
+There are now two new files in the `.ssh` directory:
+```{code-block} console
+:emphasize-lines: 5,6
+$ cd ~/.ssh
+$ ls -1
+authorized_keys
+config
+id_ed25519
+id_ed25519.pub
+known_hosts
+```
+
+The `id_ed25519` file is your private key - {bdg-warning}`It should never be shared with anyone`.
+
+The `id_ed25519.pub` file is your public key. This is the one you will copy to
+the remote machines. Le'ts do that now:
+
+```{code-block} console
+$ ssh-copy-id -i id_ed25519.pub swc-bastion
+$ ssh-copy-id -i id_ed25519.pub swc-gateway
+```
+
+The above commands will copy your public key both to the *bastion* and *gateway* nodes.
+The key will be added to the `.ssh/authorized_keys` file on each node. The
+`ssh-copy-id` command uses the configuration we previously set up
+in the `config` file to figure out how to reach the two remote machines.
+
+🎉 Congrats! You can now directly SSH into the *gateway* node without typing your password:
+```{code-block} console
+$ ssh swc-gateway
+```
+In case you want to SSH into the *bastion* node, you can do so by typing:
+```{code-block} console
+$ ssh swc-bastion
+```
+
+::: {warning}
```

Comment:
It might be worth moving this warning up, to after the code block that is after the text "'There are now two new files in the .ssh directory'". As, the instructions after this might not work if their ssh key is not saved in the correct place.